3ds Max/Max2Babylon/Exporter/BabylonExporter.IMaterialExporter.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.Drawing;
+using Autodesk.Max;
+using BabylonExport.Entities;
+using GLTFExport.Entities;
+
+namespace Max2Babylon
+{
+    
+
+    public interface IMaterialExporter
+    {
+        ClassIDWrapper MaterialClassID { get; }
+    }
+
+    public interface IBabylonMaterialExporter : IMaterialExporter
+    {
+        BabylonMaterial ExportBabylonMaterial(IIGameMaterial material);
+    }
+
+    delegate string TryWriteImageCallback(string sourceTexturePath);
+    
+    internal interface IGLTFMaterialExporter : IMaterialExporter
+    {
+        /// <summary>
+        /// Creates a GLTF material using the given GameMaterial.
+        /// </summary>
+        /// <param name="exporter">The current exporter, for export parameters like CopyTexturesToOuput.</param>
+        /// <param name="gltf">The GLTF output structure, for adding instances of classes such as GLTFSampler, GLTFImage and GLTFTexture.</param>
+        /// <param name="material">The input material matching the MaterialClassID defined by the exporter. </param>
+        /// <param name="tryWriteImageFunc">Callback function to verify images and to write images to the output folder. 
+        /// Takes the source path and the output texture name.
+        /// Returns null if the file was not written because of an error, else the output file extension.</param>
+        /// <param name="raiseMessageAction">Callback function to raise messages. Takes a message and a color.</param>
+        /// <param name="raiseWarningAction">Callback function to raise warnings.</param>
+        /// <param name="raiseErrorAction">Callback function to raise errors.</param>
+        /// <returns>The exported GLTF material.</returns>
+        GLTFMaterial ExportGLTFMaterial(BabylonExporter exporter, GLTF gltf, IIGameMaterial material, Func<string, string, string> tryWriteImageFunc,
+            Action<string, Color> raiseMessageAction, Action<string> raiseWarningAction, Action<string> raiseErrorAction);
+    }
+
+    // We require a separate struct, because the IClass_ID does not implement GetHashCode etc. to work with dictionaries
+    public struct ClassIDWrapper : IEquatable<ClassIDWrapper>
+    {
+        public static ClassIDWrapper XRef_Material = new ClassIDWrapper(0x272c0d4b, 0x432a414b);
+        public static ClassIDWrapper Advanced_Lighting_Override_Material = new ClassIDWrapper(0x2914493d, 0x6cff42f7);
+        public static ClassIDWrapper Morpher_Material = new ClassIDWrapper(0x4b9937e0, 0x3a1c3da4);
```

Comment:
Can you explain where these class ID hex values are received from? Also, It may be best to define them as constants, for the sake of reducing our magic numbers

Reply:
Also, how confident are you that these won't change with an update to Max?

Reply:
Yes we can define this constant.
About the source.
I executed this simple pyhton script in 3dsmax

```
import MaxPlus

for cd in MaxPlus.PluginManager.GetClassList().Classes:
    if cd.SuperClassId == MaxPlus.SuperClassIds.Material:
        anim = MaxPlus.Factory.CreateAnimatable(MaxPlus.SuperClassIds.Material, cd.ClassId, False)
        if anim:
            inst =  MaxPlus.Mtl._CastFrom(anim)
            if inst:
                print str(inst) + " has: " + str( cd.ClassId)
```
About the confidence:
As the class ID are generated randomly and manually I do not think Autodesk will never change any of them. This will potentially cause a lot of trouble for many third part developer

Reply:
Ok, I think going with constants is the right way to handle something like this. Mainly because if 3DS Max 2025 or something changes the ID's then for our exporter all we'd need to do is to use #defines for the exporter version